src/components/linear_parent.js

```diff
@@ -81,12 +81,49 @@ class LinearParent extends Component {
             search:event.target.value
         })
     }
+
+    removeItem=(index)=>{
+        arr.splice(index, 1);
+        n = arr.length;
+        this.setState({
+            r: n-1
+        })
+    }
+
+    addItem=()=>{
+        if(this.title.value !== ''){
+            arr.push(this.title.value);
+            n = arr.length;
+            this.setState({
+                r: n-1
+            })
+            this.title.value = '';
+        }
+    }
     render() {
         let str=this.state.start===1? <button className="button button4" style={{padding:'5px 14px'}} onClick={this.linearSearch()}>Run</button> : <button className="button button4" style={{padding:'5px 14px'}} onClick={()=>this.linearSearch()}>Run</button>
+        let items = arr.map((n, index) => {return(<span id="items" key={index}>{n} <button key={index} onClick={() => {this.removeItem(index)}} type="button" className="close"><span>&times;</span></button></span>)})
```

Comment:
Try to break jsx into small components and not merge it into one, so that it's reusable. A simple function component for it would work.